lib/utils.ts

```diff
@@ -318,12 +317,12 @@ export function clearRequireCache(options: {exclude?: string[]} = {}) {
 	}
 }
 
-export function deepClone(obj: any): any {
```

Comment:
Perhaps `mutableDeepClone` would be a more precise name. I used the `Mutable` type since in virtually all of the cases I saw, this is used to make a copy of an object and then add/mutate a property.

Since the object being cloned is often marked as `readonly`, merely `T` wouldn't suffice, as then the `readonly` would be carried over to the clone, defeating the purpose of the clone.